packages/arb-token-bridge-ui/src/components/TokenModal/TokenImportModal.tsx

```diff
@@ -129,72 +112,81 @@ export function TokenImportModal({
     }
   }, [status])
 
+  const getL1TokenData = useCallback(
+    async (eitherL1OrL2Address: string) => {
+      const addressOnL1 = await bridge?.getERC20L1Address(eitherL1OrL2Address)
+
+      if (addressOnL1) {
+        return bridge?.l1Bridge.getL1TokenData(addressOnL1)
+      } else {
+        return bridge?.l1Bridge.getL1TokenData(eitherL1OrL2Address)
+      }
+    },
+    [bridge]
+  )
+
+  const selectToken = useCallback(
+    async (_token: ERC20BridgeToken) => {
+      await token.updateTokenData(_token.address)
+      actions.app.setSelectedToken(_token)
+    },
+    [token, actions]
+  )
+
   useEffect(() => {
     if (!isOpen) {
       return
     }
 
-    if (isLoadingTokenList) {
-      return
-    }
+    const tokens = { ...tokensFromLists, ...tokensFromUser }
```

Comment:
if token is on tokensFromLists but not tokensFromUser, it ends up showing as "active" in the UI; we want to perserve and show this distinction (i.e., "we see your token but it's on a currently unimported list" etc)